src/functions.jl

```diff
@@ -142,23 +142,26 @@ end
 ################################################
 # map, collect
 
-# no explicit method needed for map(f, ::NamedDimsArray)
+# this method is only needed for Julia 1.0 in fact:
+Base.map(f, A::NamedDimsArray) = NamedDimsArray(map(f, parent(A)), names(A))
+
 for (T, S) in [
     (:NamedDimsArray, :AbstractArray),
     (:AbstractArray, :NamedDimsArray),
     (:NamedDimsArray, :NamedDimsArray),
     ]
     for fun in [:map, :map!]
 
-        @eval function Base.$fun(f, A::$T, B::$S, Cs::AbstractArray...)
+        # Here f::F where {F} is needed to avoid ambiguities in Julia 1.0
+        @eval function Base.$fun(f::F, A::$T, B::$S, Cs::AbstractArray...) where {F}
```

Comment:
It is confusing having single captial letter variables while also having single capital letter types.
Can we swap the `a::$T, b::$S, cs::AbstractArray...`